crates/bevy_asset/src/asset_server.rs

```
@@ -399,7 +403,9 @@ impl AssetServer {
         let path = path.as_ref();
         if !self.server.asset_io.is_directory(path) {
             return Err(AssetServerError::AssetFolderNotADirectory(
-                path.to_str().unwrap().to_string(),
+                path.to_str()
+                    .expect("Could not convert path to string.")
```

Seems like this should only fail if the path has invalid utf8.